docs/v1/P4Runtime-Spec.mdk

```diff
@@ -622,8 +622,9 @@ reasons:
    assigned to each role, is **out-of-scope** of this document.
 
 2. Redundancy and fault tolerance: Supporting multiple controllers allows having
-   one or more standby slave controllers, which take over controlling the
-   devices in case the master controller goes offline.
+   one or more standby slave controllers. These can already have a connection
+   open, and can read the switch state while being a slave, which can help them
+   become master more quickly.
```

Comment:
Is it actually true that a new master can become up to date with the current state more quickly given today's P4Runtime protocol, unless it has some way of knowing which of the data it read before it became master is still valid, and which is stale?

i.e. if P4Runtime had a per-table "content version id" number that was monotonically increasing whenever any of its contents were changed, and this value could be read by a client without reading all of the table state, and it came with a read of an entire table's state, then a new master could quickly read that value for all tables, compare it against what it has now, and then only read the contents of tables that it did not have the latest version of.

With today's protocol, it seems that no matter how many reads a slave may have performed before, it must assume that any or all of it could change by the time it becomes the master, because of writes issued by the previous master since the last time the slave did a read.

Reply:
I agree with this point. I'm not sure what we should do about it. It seems like a major change, but it might make sense to remove the notion of a standby controller.

Reply:
Would it be sufficient to remove "Redundancy and fault tolerance" as a goal?  It really seems like we are falling short of that right now.

Reply:
I remember a conversation with @wmohsin where he mentioned the potential cost of establishing a connection from a new controller to the switch (cost to discover a route between controller and switch). Maybe that alone justifies having standby controllers?
There is also what we discussed earlier today. Notifying standby controllers that the master has disconnected (assuming the master cannot itself notify the consensus layer for some reason) may enable faster recovery.

Reply:
In any case, we should definitely remove "and can read the switch state while being a slave", but as I wrote above there may be some value in already having an open connection

Reply:
Done.